systems/plants/collision/BulletModel.cpp

```diff
@@ -202,6 +204,67 @@ namespace DrakeCollision
     return id;
   }
 
+  bool BulletModel::potentialCollisionPoints(bool use_margins, vector<PointPair> & potential_collisions)
+  {
+    BulletCollisionWorldWrapper& bt_world = getBulletWorld(use_margins);
+    BulletResultCollector c;
+    bt_world.bt_collision_world->performDiscreteCollisionDetection();
+    size_t numManifolds = bt_world.bt_collision_world->getDispatcher()->getNumManifolds();
+    for (size_t i = 0; i < numManifolds; i++)
+    {
+      btPersistentManifold* contactManifold =  bt_world.bt_collision_world->getDispatcher()->getManifoldByIndexInternal(i);
+      const btCollisionObject* obA = contactManifold->getBody0();
+      const btCollisionObject* obB = contactManifold->getBody1();
+
+      auto elementA = static_cast<Element*>(obA->getUserPointer());
+      auto elementB = static_cast<Element*>(obB->getUserPointer());
+
+      DrakeShapes::Shape shapeA = elementA->getShape();
+      DrakeShapes::Shape shapeB = elementB->getShape();
+
+      ElementId idA = elementA->getId();
+      ElementId idB = elementB->getId();
+
+      double marginA = 0;
+      double marginB = 0;
+
+      if (shapeA ==  DrakeShapes::MESH || shapeA ==  DrakeShapes::BOX) { 
+        marginA = obA->getCollisionShape()->getMargin();
+      }
+      if (shapeB == DrakeShapes::MESH || shapeB == DrakeShapes::BOX) { 
+        marginB = obB->getCollisionShape()->getMargin();
+      }
+      int numContacts = contactManifold->getNumContacts();
+
+      for (int j=0;j<numContacts;j++)
+      {        
+        btManifoldPoint& pt = contactManifold->getContactPoint(j);
+        const btVector3& normalOnB = pt.m_normalWorldOnB;
+        const btVector3& pointOnAinWorld = pt.getPositionWorldOnA() + normalOnB * marginA;
+        const btVector3& pointOnBinWorld = pt.getPositionWorldOnB() - normalOnB * marginB;
+        const btVector3 pointOnElemA = obA->getWorldTransform().invXform(pointOnAinWorld);
+        const btVector3 pointOnElemB = obB->getWorldTransform().invXform(pointOnBinWorld);
+
+        Vector4d pointOnA_1,  pointOnB_1;
+
+        pointOnA_1 << toVector3d(pointOnElemA), 1.0;
+        pointOnB_1 << toVector3d(pointOnElemB), 1.0;
+
+        Vector3d pointOnA;
+        Vector3d pointOnB;
+
+        pointOnA << elements[idA]->getLocalTransform().topRows(3) * pointOnA_1;
+        pointOnB << elements[idB]->getLocalTransform().topRows(3) * pointOnB_1;
+
+        c.addSingleResult(idA, idB, pointOnA, pointOnB, toVector3d(normalOnB), static_cast<double>(pt.getDistance()) + marginA + marginB);
```

Comment:
code formatting in this file: you're using camelcase for variables; should be underscores. I'd also prefer camelcase, but it's in the style guide.